evm/vm/base.py

```diff
@@ -156,6 +159,13 @@ def mine_block(self, *args, **kwargs):
         Mine the current block. Proxies to self.pack_block method.
         """
         block = self.block
+        tx_root_hash, tx_kv_nodes = make_trie_root_and_nodes(block.transactions)
+        receipt_root_hash, receipt_kv_nodes = make_trie_root_and_nodes(self.receipts)
+        trie_data = merge(tx_kv_nodes, receipt_kv_nodes)
```

Comment:
Probably unimportant: for performance optimization, this line could be removed, and just add them separately:
```python
self.chaindb.persist_trie_data_dict(tx_kv_nodes)
self.chaindb.persist_trie_data_dict(receipt_kv_nodes)
```